paimon-common/src/main/java/org/apache/paimon/fileindex/bitmap/BitmapFileIndex.java

```diff
@@ -251,18 +280,20 @@ private void readInternalMeta(DataType dataType) {
                 this.valueMapper = getValueMapper(dataType);
                 try {
                     seekableInputStream.seek(headStart);
-                    this.version = seekableInputStream.read();
-                    if (this.version > VERSION_1) {
+                    int version = seekableInputStream.read();
+                    if (version == VERSION_1) {
+                        this.bitmapFileIndexMeta = new BitmapFileIndexMeta(dataType, options);
+                        this.bitmapFileIndexMeta.deserialize(seekableInputStream);
```

Comment:
If we know the length of the bitmap, can we just use `RoaringBitmap32.deserialize(ByteBuffer buffer)`?